CMakeLists.txt

```diff
@@ -466,27 +474,27 @@ feature_option(static_runtime "build libtorrent with static runtime" OFF)
 find_public_dependency(Threads REQUIRED)
 
 if(CMAKE_CXX_COMPILER_ID MATCHES Clang)
-	add_compile_options(
-		-Weverything
-		-Wno-documentation
-		-Wno-c++98-compat-pedantic
-		-Wno-padded
-		-Wno-global-constructors
-		-Wno-exit-time-destructors
-		-Wno-weak-vtables
-		-Wno-return-std-move-in-c++11
-		-Wno-unknown-warning-option)
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} \
+		-Weverything \
+		-Wno-documentation \
+		-Wno-c++98-compat-pedantic \
+		-Wno-padded \
+		-Wno-global-constructors \
+		-Wno-exit-time-destructors \
+		-Wno-weak-vtables \
+		-Wno-return-std-move-in-c++11 \
+		-Wno-unknown-warning-option")
```

Comment:
I changed it because `add_compile_options` adds options also for C code, which breaks the compilation of libdatachannel's dependencies. Since the options were added globally before anyway I changed to `CMAKE_CXX_FLAGS` to make them exclusive to C++.

Reply:
oh, I see. I didn't realize at first that this adds *global* compiler flags. I'm a bit surprised they aren't just added to the `libtorrent` target.
@zeule do you have any thoughts on this?

Reply:
No, I don't know an elegant way to achieve what we want, because we want those flags to be defined for the libtorrent target and the tests, but we can't make them public libtorrent flags because we don't want them to escape into the libtorrent interface (via pkg-config or cmake targets).

These two solutions seem to be least ugly to me:
1. Add a dummy target with these options in its INTERFACE flags. Link it privately to libtorrent and the tests.
2. Add these flags to the libtorrent target as PUBLIC, but hide them under `$<BUILD_INTERFACE:` generator expression, so that they will not escape to the exported target. 

The second one needs testing for the pkg-config generator, but otherwise is, probably, the best way to go.

Reply:
option two seems risky to me. it would mean anyone adding libtorrent as a cmake dependency would get all these flags too, right?

Reply:
Yes, if they reference the target from a build tree.